fractions/javaee/ejb/src/main/java/org/wildfly/swarm/ejb/EJBFraction.java

```diff
@@ -56,32 +53,54 @@ public EJBFraction applyDefaults() {
         threadPoolSettings.put("time", "100");
         threadPoolSettings.put("unit", "MILLISECONDS");
 
-        defaultStatefulBeanAccessTimeout(5000L)
-                .defaultSingletonBeanAccessTimeout(5000L)
+        this
+                .defaultSlsbInstancePool("slsb-strict-max-pool")
+                .defaultStatefulBeanAccessTimeout(5000L)
                 .defaultSfsbCache("simple")
+                .defaultSfsbPassivationDisabledCache("simple")
+                .defaultSingletonBeanAccessTimeout(5000L)
+                .defaultResourceAdapterName(SwarmProperties.propertyVar("ejb.resource-adapter-name", "activemq-ra.rar"))
+                .defaultMdbInstancePool("mdb-strict-max-pool")
+                .strictMaxBeanInstancePool("slsb-strict-max-pool", pool -> {
+                    pool.deriveSize(StrictMaxBeanInstancePool.DeriveSize.FROM_WORKER_POOLS);
+                    pool.timeout(5L);
+                    pool.timeoutUnit(StrictMaxBeanInstancePool.TimeoutUnit.MINUTES);
+                })
+                .strictMaxBeanInstancePool("mdb-strict-max-pool", pool -> {
+                        pool.deriveSize(StrictMaxBeanInstancePool.DeriveSize.FROM_CPU_COUNT);
+                        pool.timeout(5L);
+                        pool.timeoutUnit(StrictMaxBeanInstancePool.TimeoutUnit.MINUTES);
+                })
+                .cache("simple")
+                // ideally, the distributable cache and the infinispan passivation store would be defined
+                // in InfinispanCustomizer, but that only applies after YAML is processed
+                .cache("distributable", cache -> {
+                    cache.alias("passivating");
+                    cache.alias("clustered");
+                    cache.passivationStore("infinispan");
+                })
+                .passivationStore("infinispan", store -> {
+                    store.cacheContainer("ejb");
+                    store.maxSize(10000);
+                })
+                .asyncService(async -> {
+                    async.threadPoolName(DEFAULT);
+                })
+                .timerService(timer -> {
+                    timer.threadPoolName(DEFAULT);
+                    timer.defaultDataStore("default-file-store");
+                    timer.fileDataStore("default-file-store", store -> {
+                        store.path("timer-service-data");
+                        store.relativeTo("jboss.server.data.dir");
+                    });
+                })
+                .threadPool(DEFAULT, threadPool -> {
+                    threadPool.maxThreads(10);
+                    threadPool.keepaliveTime(threadPoolSettings);
+                })
                 .defaultSecurityDomain("other")
                 .defaultMissingMethodPermissionsDenyAccess(true)
-                .logSystemExceptions(true)
```

Comment:
any particular reason for changing the sequencing?

makes it hard to know what's actually changed

Reply:
I just rewrote this from scratch, following the order from `standalone-full.xml`, because there were too many missing things. I'll check if I can make it more review-friendly.